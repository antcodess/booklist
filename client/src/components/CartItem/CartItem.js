import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import { addItem, removeItem } from '../../redux/cart/cart.actions';

import './CartItem.scss';

export const CartItem = ({ item, history, addItem, removeItem }) => {
  const { title, imageUrl, price, quantity } = item;

  return (
    <div className='cart-item'>
      <img src={imageUrl} alt='item' onClick={() => {
        history.push(`/${item.routeName}`)
      }}/>
      <div className='item-details'>
        <span className='name' onClick={() => {
          history.push(`/${item.routeName}`)
        }}>{title}</span>
        <span className='quantity'>
        <div className='arrow' id='removeItemArrow' onClick={() => removeItem(item)}>&#10094;</div>
        <span className='value'>{quantity} x ${price}</span>
        <div className='arrow' id='addItemArrow' onClick={() => addItem(item)}>&#10095;</div>
      </span>
      </div>
    </div>
  );
};

export const mapDispatchToProps = dispatch => ({
  addItem: item => dispatch(addItem(item)),
  removeItem: item => dispatch(removeItem(item))
});

export default connect(null, mapDispatchToProps)(withRouter(CartItem));

