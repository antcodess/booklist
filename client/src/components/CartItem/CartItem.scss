.cart-item {
  width: 100%;
  display: flex;
  height: 70px;
  margin-top: 15px;

  img {
    width: 30%;
  }
  
  .item-details {
    width: 68%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    .name {
      font-family: "Cera GR Bold";
      font-size: 1rem;
    }
    
    .quantity {
      display: flex;
      justify-content: center;
      margin-top: 2px;
  
      .arrow {
        cursor: pointer;
      }
  
      .value {
        margin: 2px 10px;
        font-size: 0.94rem;
      }
    }
  }
}