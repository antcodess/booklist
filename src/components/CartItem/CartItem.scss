.cart-item {
  width: 100%;
  display: flex;
  height: 70px;
  margin-top: 15px;

  img {
    width: 30%;
  }
  
  .item-details {
    width: 70%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    .name {
      font-family: "Cera GR Bold";
      font-size: 1rem;
    }
    .price {
      font-family: "Cera GR Light";
      font-size: 0.94rem;
    }
  }
}