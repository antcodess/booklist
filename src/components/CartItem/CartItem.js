import React from 'react';
import { withRouter } from 'react-router-dom';

import './CartItem.scss';

const CartItem = ({ item, history }) => {
  const { title, imageUrl, price, quantity } = item;
  return (
    <div className='cart-item' onClick={() => history.push(`/${item.routeName}`)}>
      <img src={imageUrl} alt='item' />
      <div className='item-details'>
        <span className='name'>{title}</span>
        <span className='price'>{quantity} x ${price}</span>
      </div>
    </div>
  );
};

export default withRouter(CartItem);