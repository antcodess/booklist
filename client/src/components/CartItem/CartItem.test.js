import React from 'react';
import { shallow } from 'enzyme';
import { CartItem } from './CartItem';
import { mapDispatchToProps } from './CartItem';

describe('CartItem Component', () => {
  let wrapper;
  let mockItem;
  let mockHistory;
  let mockAddItem;
  let mockRemoveItem;
  
  beforeEach(() => {

    mockItem = { id: 1 };
    mockHistory = { push: jest.fn() };
    mockAddItem = jest.fn();
    mockRemoveItem = jest.fn();

    const mockProps = {
      item: mockItem, 
      history: mockHistory, 
      addItem: mockAddItem, 
      removeItem: mockRemoveItem
    };

    wrapper = shallow(<CartItem { ...mockProps } />);
  });

  it('should render CartItem Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call history.push when clicked', () => {
    wrapper.find('img').simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
  });

  it('should call history.push when clicked 2', () => {
    wrapper.find('.name').simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
  });

  it('should call removeItem when clicked', () => {
    wrapper.find('#removeItemArrow').simulate('click');
    expect(mockRemoveItem).toHaveBeenCalled();
  });

  it('should call addItem when clicked', () => {
    wrapper.find('#addItemArrow').simulate('click');
    expect(mockAddItem).toHaveBeenCalled();
  });

  it('should dispatch addItem action when clicked', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).addItem(mockItem);
    expect(dispatch.mock.calls[0][0]).toEqual(
      { type: 'ADD_ITEM', payload: { id: 1 } }
    );
  });

  it('should dispatch removeItem action when clicked', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).removeItem(mockItem);
    expect(dispatch.mock.calls[0][0]).toEqual(
      { type: 'REMOVE_ITEM', payload: { id: 1 } }
    );
  });
});