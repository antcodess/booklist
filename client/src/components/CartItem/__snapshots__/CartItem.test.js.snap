// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem Component should render CartItem Component 1`] = `ShallowWrapper {}`;
